src/Core/Tool/Signer.php

```diff
@@ -27,42 +27,19 @@ public function sign($fullUrl, $json)
     public function computeSignature($url, $json)
     {
         $url = $url . $json;
+
         // This function calculates the HMAC hash of the data with the key
         // passed in
         // Note: hash_hmac requires PHP 5 >= 5.1.2 or PECL hash:1.1-1.5
         // Or http://pear.php.net/package/Crypt_HMAC/
         return base64_encode(hash_hmac("sha256", $url, $this->authToken, true));
     }
 
-    public function validate($expectedSignature, $url, $data = array())
+    public function validate($expectedSignature, $url, $data = "")
     {
-        return self::compare(
-            $this->computeSignature($url, json_encode($data)),
+        return hash_equals(
```

Comment:
@willdoran I'd really like you to check this. I've dropped the DIY comparison function in favour of PHP's built int.
The docs mention:

> Both arguments must be of the same length to be compared successfully. When arguments of differing length are supplied, FALSE is returned immediately and the length of the known string may be leaked in case of a timing attack.

However from my reading of `compare()` below, it looks like it exited out early if strlen was different so would has a similar timing vulnerability.

Reply:
I think you're correct, the one below does have the same issue, I'm not sure why the note says that it doesn't. We can either go with the hash_equals and note the issue or we can implement one of the time insensitive versions.